Add "execution" and "consensus" terms to glossary (#5192)
ethereum/ethereum-org-website: src/content/glossary/index.md

When numerous nodes (usually most nodes on the network) all have the same blocks in their locally validated best blockchain. Not to be confused with [consensus rules](#consensus-rules).

### consensus client {consensus-client}

> After the merge to [proof-of-stake](#pos), consensus about the true head of the blockchain will be governed by a new set of clients running on the Beacon Chain. These are "consensus clients". They do not participate in validating transactions or executing state transitions, they request this to be done by [execution clients](#execution-client).

Reviewer: ```suggestion
After the merge to [proof-of-stake](#pos), consensus about the true head of the blockchain will be governed by a new set of clients running on the Beacon Chain known as "consensus clients". Consensus clients do not participate in validating transactions or executing state transitions. They request this to be done by [execution clients](#execution-client).
```

> After the merge to [proof-of-stake](#pos), consensus about the true head of the blockchain will be governed by a new set of clients running on the Beacon Chain known as "consensus clients". Consensus clients do not participate in validating transactions or executing state transitions. They request this to be done by [execution clients](#execution-client).

Reviewer 2: > After the merge to [proof-of-stake](#pos), consensus about the true head of the blockchain will be governed by a new set of clients running on the Beacon Chain known as "consensus clients".

While this is true, I would add a sentence before to explain what consensus clients are before going into The Merge, given they exist today. Perhaps something like "Consensus clients (a.k.a. Beacon Chain client) run Ethereum's proof of stake consensus algorithm." I would also suggest naming a few of them to make it clear to readers "what" they are. 

Also a few small nits:
- "the merge to proof of stake" -> "the transition to proof of stake" OR "the merge": we are moving from PoW to PoS *by* merging the changes, so it's a bit odd to say "the merge to PoS";
- "true head" -> "head": by saying you are coming to consensus on it, you imply it's "true";
- "will be governed" -> "will be determined": there isn't really governance involved here, risks confusing people;
- "a new set of clients": not quite accurate, given they are already existent. Assuming you added the earlier sentence I proposed, you can just use "these clients". 

> Consensus clients do not participate in validating transactions or executing state transitions. They request this to be done by [execution clients](#execution-client).

Two small nits here: 

- "validating" -> "validating/broadcasting": txn gossip will be on the EL;
- "They request this to be done" -> "This is done by": while it's technically true the consensus client makes a request to the execution layer client, this phrasing implies some sort of superiority, which I think we want to avoid.

### consensus layer {consensus-layer}

> The network of Ethereum [consensus clients](#consensus-client).

Reviewer 3: ```suggestion
Ethereum's consensus layer is the network of [consensus clients](#consensus-client).
```

In the context of cryptography, lack of predictability or level of randomness. When generating secret information, such as [private keys](#private-key), algorithms usually rely on a source of high entropy to ensure the output is unpredictable.

### execution client {#execution-client}

> After the merge to [proof-of-stake](#pos) existing Ethereum mainnet clients will continue to host the [Ethereum Virtual Machine](#evm), validate transactions and execute state transitions but will not participate in consensus. These clients will therefore be referred to as "execution clients".

Reviewer: ```suggestion
After the merge to [proof-of-stake](#pos), existing Ethereum Mainnet clients will continue to host the [Ethereum Virtual Machine](#evm), validate transactions and execute state transitions but will not participate in consensus. These clients will therefore be referred to as "execution clients".
```

Author: looks good - thanks!

> After the merge to [proof-of-stake](#pos), existing Ethereum Mainnet clients will continue to host the [Ethereum Virtual Machine](#evm), validate transactions and execute state transitions but will not participate in consensus. These clients will therefore be referred to as "execution clients".

Reviewer 2: I think this one isn't 100% accurate. My suggestion:

> Execution clients (f.k.a. "Eth1 clients"), such as Besu, Erigon, go-ethereum, Nethermind, are tasked with processing and broadcasting transactions, as well as with managing Ethereum's state. They run the computations for each transaction in the Ethereum Virtual Machine to ensure that the rules of the protocol are followed. Today, they also handle proof of work consensus. After the transition to proof of stake, they will delegate this to consensus clients.

Author: Thanks a lot for improving this @timbeiko!
